src/pet.js

```diff
@@ -0,0 +1,70 @@
+const INITIAL_AGE = 0;
+const BASE_HUNGER = 0;
+const MAX_FITNESS = 10;
+
+function Pet(name) {
+  this.name = name;
+  this.age = INITIAL_AGE;
+  this.hunger = BASE_HUNGER;
+  this.fitness = MAX_FITNESS;
+  this.children = [];
+}
+
+Pet.prototype = {
+  get isAlive() {
+    return this.age < 30 && this.hunger < 10 && this.fitness > 0;
+  },
+};
+
+Pet.prototype.growUp = function () {
+  if (!this.isAlive) {
+    throw new Error("Your pet is no longer alive :(");
+  }
+  this.age += 1;
+  this.hunger += 5;
+  this.fitness -= 3;
+};
+
+Pet.prototype.walk = function () {
+  if (!this.isAlive) {
+    throw new Error("Your pet is no longer alive :(");
+  }
+  if (this.fitness + 4 <= MAX_FITNESS) {
+    this.fitness += 4;
+  } else {
+    this.fitness = MAX_FITNESS;
+  }
```

Comment:
This could be refactored a bit. In essence you are adding a value to the existing fitness, without allowing it go over a set limit. You can do this with Math.min, which will return the lowest of the values passed to it:
```
this.fitness = Math.min(this.fitness + 4, MAX_FITNESS)
```